.github/workflows/publish.yaml

```diff
@@ -0,0 +1,35 @@
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+      # Releases
+      - '[0-9]+.[0-9]+.[0-9]+'
+      # Release Candidates
+      - '[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+
+permissions:
+  contents: write
+  id-token: write
+  pull-requests: write
+
+jobs:
+  create-sbom-release-asset:
```

Comment:
Aren't we uploading an sbom for each PR & master build? is it necessary to do it again for the tag event?

Reply:
I _think_ it is because the tag is a separate entity from the commit (even though the tag is mapped to a commit), but I'm really not sure.

Reply:
To be clear, I think this will ensure that the SBOM artifact is associated with the Github release and shows up under the release's assets (https://github.com/Workiva/dpx/releases).